OpenRA.Mods.Common/AI/BaseBuilder.cs

```diff
@@ -153,33 +211,69 @@ ActorInfo ChooseBuildingToBuild(ProductionQueue queue)
 			if (!ai.HasAdequateProc() || !ai.HasMinimumProc())
 			{
 				var refinery = GetProducibleBuilding("Refinery", buildableThings);
-				if (refinery != null)
+				if (refinery != null && HasSufficientPowerForActor(refinery))
 				{
 					HackyAI.BotDebug("AI: {0} decided to build {1}: Priority override (refinery)", queue.Actor.Owner, refinery.Name);
 					return refinery;
 				}
+
+				if (power != null && refinery != null && !HasSufficientPowerForActor(refinery))
+				{
+					HackyAI.BotDebug("{0} decided to build {1}: Priority override (would be low power)", queue.Actor.Owner, power.Name);
+					return power;
+				}
 			}
 
 			// Make sure that we can can spend as fast as we are earning
 			if (ai.Info.NewProductionCashThreshold > 0 && playerResources.Resources > ai.Info.NewProductionCashThreshold)
 			{
 				var production = GetProducibleBuilding("Production", buildableThings);
-				if (production != null)
+				if (production != null && HasSufficientPowerForActor(production))
 				{
 					HackyAI.BotDebug("AI: {0} decided to build {1}: Priority override (production)", queue.Actor.Owner, production.Name);
 					return production;
 				}
+
+				if (power != null && production != null && !HasSufficientPowerForActor(production))
+				{
+					HackyAI.BotDebug("{0} decided to build {1}: Priority override (would be low power)", queue.Actor.Owner, power.Name);
+					return power;
+				}
+			}
+
+			// Make sure that we can can spend as fast as we are earning
```

Comment:
The comment doesn't match what this is doing.  You can increase spending by building any new production structure - there's nothing special about naval.